File: TEMA2/python/randomnumber.py
from random import randint

def rand(n,odd="",even=""):
    for i in range(n):
        r=randint(1,100)
        if r%2==0:
            even+=f'{r} '
        else:
            odd+=f'{r} '
    return f'Pares: {even} \nImpares: {odd}'

n=int(input('Introduzca el limite: '))
print(rand(n))

File: TEMA2/typescript/randomnumber.ts
import readline from "node:readline/promises";
import { stdin as input, stdout as output } from "node:process";
const rl = readline.createInterface({ input, output });
const {floor,random}=Math;

function rand(n: number,odd: string="", even: string = ""): string {
    for (let i=0;i<n;i++){
        const r = floor(random()*100)+1;
        if (r % 2 === 0) {
            even += `${r} `;
        } else {
            odd += `${r} `;
        }
    }
    return `Pares: ${even}\nImpares: ${odd}`;
}

const n = Number(await rl.question("Introduzca el limite: "));
console.log(rand(n));
rl.close();